File: src/01-easy/ascii-art/solution.swift
let L=Int(readLine()!)!,H=Int(readLine()!)!
let T=Array(readLine()!.uppercased().unicodeScalars)
var rows=[String]();for _ in 0..<H{rows.append(readLine()!)}
for i in 0..<H{
    var line="";let arr=Array(rows[i])
    for c in T{
        var idx=Int(c.value)-65;if idx<0||idx>25{idx=26}
        line+=String(arr[idx*L..<idx*L+L])
    }
    print(line)
}

File: src/01-easy/ascii-art/solution.lua
local L=tonumber(io.read())
local H=tonumber(io.read())
local T=io.read():upper()
local rows={};for i=1,H do rows[i]=io.read() end
for i=1,H do
    local line=""
    for j=1,#T do
        local idx=T:byte(j)-65
        if idx<0 or idx>25 then idx=26 end
        line=line..rows[i]:sub(idx*L+1,idx*L+L)
    end
    print(line)
end
